source/Grid/Grid.js

```diff
@@ -740,10 +748,11 @@ export default class Grid extends Component {
     // This causes a series of rapid renders that is slow for long lists.
     // We can avoid that by doing some simple bounds checking to ensure that scrollTop never exceeds the total height.
     const { height, width } = this.props
+    const scrollbarSize = getScrollbarSize()
     const totalRowsHeight = this._getTotalRowsHeight()
     const totalColumnsWidth = this._getTotalColumnsWidth()
-    const scrollLeft = Math.min(totalColumnsWidth - width, event.target.scrollLeft)
-    const scrollTop = Math.min(totalRowsHeight - height, event.target.scrollTop)
+    const scrollLeft = Math.min(totalColumnsWidth - width + scrollbarSize, event.target.scrollLeft)
+    const scrollTop = Math.min(totalRowsHeight - height + scrollbarSize, event.target.scrollTop)
```

Comment:
I agree it doesn't need to be recalcualted over and over, luckily the helper does already cache the result, so we don't need to worry about it here (https://github.com/react-bootstrap/dom-helpers/blob/master/src/util/scrollbarSize.js). I should have noted that, sorry!

I can still move it to a private property if that is helpful, it may just be clearer for folks looking at the code later.